src/Symfony/Component/ObjectMapper/MapperMetadataFactoryInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ObjectMapper;
+
+use Symfony\Component\ObjectMapper\Attributes\Map;
+
+/**
+ * Factory to create Mapper metadata.
+ *
+ * @experimental
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+interface MapperMetadataFactoryInterface
```

Comment:
Could this interface be `@internal` instead of `@experimental` ? i do believe this component should reduce as much as possible public API for a start and open it on user wish given a concret behavior